reason-parser/src/reason_parser.mly

```diff
@@ -161,6 +161,13 @@ let make_ghost_loc loc = {
 
 let ghloc ?(loc=dummy_loc ()) d = { txt = d; loc = (make_ghost_loc loc) }
 
+let rev_list_extract_loc prj lst =
+  let last = List.hd lst in
+  let lst = List.rev lst in
+  let head = List.hd lst in
+  let loc = mklocation (prj head).loc_start (prj last).loc_end in
+  (lst, loc)
+
```

Comment:
@SanderSpies: this is an implementation of the idea I suggested. Do you think it could be a problem with handling of whitespaces? How so?